src/resource/solidDataset.ts

```diff
@@ -49,28 +45,159 @@ import {
   getResourceInfo,
   isContainer,
   FetchError,
+  responseToResourceInfo,
+  getContentType,
 } from "./resource";
 import {
-  internal_cloneResource,
   internal_isUnsuccessfulResponse,
   internal_parseResourceInfo,
 } from "./resource.internal";
-import { thingAsMarkdown, getThingAll, getThing } from "../thing/thing";
+import { thingAsMarkdown, getThing, getThingAll } from "../thing/thing";
 import {
   internal_getReadableValue,
-  internal_toNode,
   internal_withChangeLog,
 } from "../thing/thing.internal";
 import { getIriAll } from "../thing/get";
 import { normalizeServerSideIri } from "./iri.internal";
+import {
+  addRdfJsQuadToDataset,
+  freeze,
+  getChainBlankNodes,
+  getLocalNodeName,
+  isLocalNodeIri,
+  toRdfJsQuads,
+} from "../rdf.internal";
 
 /**
  * Initialise a new [[SolidDataset]] in memory.
  *
  * @returns An empty [[SolidDataset]].
  */
 export function createSolidDataset(): SolidDataset {
-  return dataset();
+  return freeze({
+    type: "Dataset",
+    graphs: {
+      default: {},
+    },
+  });
+}
+
+export type Parser = {
+  onQuad: (onQuadCallback: (quad: Quad) => void) => void;
+  onError: (onErrorCallback: (error: unknown) => void) => void;
+  onComplete: (onCompleteCallback: () => void) => void;
+  parse: (source: string, resourceInfo: WithServerResourceInfo) => void;
+};
```

Comment:
Out of curiosity, besides n3, have you considered any parser that could instantiate this interface ?

Reply:
One thing to note is that n3 does not conform to this interface. Instead, I have considered other parsers in the sense that theoretically, you should be able to add a wrapper similar to the one I wrote for n3 to make other parsers work as well. However, I would like to actually write those for at least JSON-LD and a different Turtle parser before making this part of our official public API.

(So add the functionality now, but list it in the changelog and API docs later.)

Reply:
Not-yet-public interfaces hidden in 5e5a5e4a2410c967c4a5cce63cbe1b8365949c75.